gluten-iceberg/src/main/scala/org/apache/iceberg/spark/source/GlutenIcebergSourceUtil.scala

```diff
@@ -50,6 +51,7 @@ object GlutenIcebergSourceUtil {
     }
   }
 
+  @nowarn
```

Comment:
What's the warning message? Can we fix the warning?

Reply:
`org.apache.iceberg.ContentFile::path()` is marked as `deprecated`. Without introducing a shim, it's hard to fix properly, so we'll suppress the warning for now.

```scala
        asFileScanTask(tasks.toList).foreach {
          task =>
            paths.add(
              BackendsApiManager.getTransformerApiInstance
                .encodeFilePathIfNeed(task.file().path().toString))  // => deprecated
``` 

Additionally, `@nowarn` can only be added before method names; it cannot be applied before individual statements.